src/Jarvis.Core/Diagnostics/TimedOperationExtensions.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to Spectre Systems AB under one or more agreements.
+// Spectre Systems AB licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Jarvis.Core.Diagnostics
+{
+    public static class TimedOperationExtensions
+    {
+        public static IDisposable TimedOperation(this IJarvisLog logger, string messageTemplate,
+            params object[] propertyValues)
+        {
+            return logger.TimedOperation(LogLevel.Information, messageTemplate, propertyValues);
+        }
+
+        public static IDisposable TimedOperation(this IJarvisLog logger, LogLevel level, string messageTemplate,
+            params object[] propertyValues)
+        {
+            return logger.TimedOperation(level, level, messageTemplate, propertyValues);
+        }
+
+        public static IDisposable TimedOperation(this IJarvisLog logger, LogLevel completionLevel, LogLevel cancelledLevel, string messageTemplate,
+            params object[] propertyValues)
+        {
+            return new Operation(logger, messageTemplate, propertyValues, completionLevel, cancelledLevel);
+        }
+    }
+
+    public class Operation : IDisposable
```

Comment:
We should put this class in it's own file and make it sealed. It can also be made internal.

Reply:
Perhaps we could rename this to `TimedScope` or similar.

Reply:
There are subtle, but important differences between a Scope and the Timed Operation: a scope is something in which log entries are enriched with property/properties. The TimedOperation's responsibility is to produce an log entry of an performed operation. Usage-wise they look similar, are disposable typically in a `using` statement, but I see them as two conceptually different things.

Reply:
I agree that it should be moved to it's own file and sealed :ok_hand:

Reply:
I should also say that the naming of `Operation` and parts of the API is heavily influenced by [serilog-timings](https://github.com/nblumhardt/serilog-timings) that IMHO has one of the nicest APIs for timings.

Reply:
Ok, can we rename it to `TimedOperation` at least? Not a fan of the generic `Operation` name.

Reply:
@pardahlman Ping